### ISSUE TITLE
Unable to dismiss keyboard for the Terra Search Field

### ISSUE THREAD

**Issue by ccehlers** [cerner/terra-core#3572]
# Bug Report

## Description
On a Mobile view when using the Terra Search Field, a user is unable to dismiss the keyboard when the enter key is pressed, leaving the user stuck on the screen.  We've had consistent client feedback of this being defective.  

## Steps to Reproduce

1. Enter a mobile view with the Terra Search Field or Terra Input Field and the keyboard present
2. Select the 'Enter' button on the keyboard
3. Keyboard does not dismiss and leaves the user confused on how to access the rest of the page content.

## Additional Context / Screenshots
![image](https://user-images.githubusercontent.com/12070453/151453349-c0250bea-1ee6-4a84-87d6-c901648653b6.png)

## Expected Behavior
Upon hitting the Enter key, the user again sees the rest of the page content and the keyboard dismisses.  The expectation would be to have a UX that manages both to be intuitive to the user as well as meet Accessibility requirements.  

## Possible Solution
The challenge with this will be Accessibility concerns, as the reason it behaves this way is due to the need for the field to always be in context and not blur out the field to dismiss the keyboard.  However this is behavior is also consistently being reported as defective by our clients within Nursing workflows on mobile devices.

## Environment
terra-search-field (3.74.0) + terra-form-input (4.17.0)
Chrome for Android, Safari for iOS - any version
node 10/npm 6
webpack 4
Mobile Android + iOS

## Notes
This can be easily reproduced by visiting one of the terra component documentation pages in a mobile browser (ex. https://engineering.cerner.com/terra-ui/components/terra-form-input/form-input/input)

## @ Mentions
@scottwilmarth

**Comment by benbcai** [comment]
Logged UXPLATFORM-6428 to track this issue.
